src/app/container/container.component.ts

```diff
@@ -98,7 +99,30 @@ export class ContainerComponent extends Entry {
     this.extendedTool$ = this.extendedDockstoreToolQuery.extendedDockstoreTool$;
 
     this._toolType = 'containers';
-    this.redirectAndCallDiscourse('/my-tools');
+    this.redirectToCanonicalURL('/my-tools');
+  }
+
+  ngAfterViewInit() {
+    if (this.publicPage) {
+      this.toolQuery.tool$.pipe(takeUntil(this.ngUnsubscribe)).subscribe(
+        tool => {
+
+        if (tool && tool.topicId) {
```

Comment:
This is probably not a scenario, but can either these be set and then not set as the events fire? If so, the DiscourceEmbed would end up pointing to an older topic.

Reply:
I do not think this would happen, but I don't want to say with absolute certainty that it can't. Unfortunately I can't really do much testing until this is one staging.